### PR TITLE
Fix/ota 4941/installed version matching

### PR DESCRIPTION
**Pull request by patrickvacek**

### CODE REVIEW

File: src/libaktualizr/uptane/imagerepository.cc [hunk 1]
```diff
@@ -115,7 +115,7 @@ void ImageRepository::fetchTargets(INvStorage& storage, const IMetadataFetcher&
   verifyTargets(image_targets, false);
 
   if (local_version > remote_version) {
-    throw Uptane::SecurityException(RepositoryType::IMAGE, "Mismatched target versions");
+    throw Uptane::SecurityException(RepositoryType::IMAGE, "Rollback attempt");
```
Comment (zabbal) [inline src/libaktualizr/uptane/imagerepository.cc:118]: That looks like smth which deserves big bold red error in the log output. Is this done where this exception is caught?

File: src/libaktualizr/package_manager/packagemanagerfake.cc [hunk 2]
```diff
@@ -110,6 +110,7 @@ bool PackageManagerFake::fetchTarget(const Uptane::Target& target, Uptane::Fetch
     return false;
   }
 
+  // TODO(OTA-4939): Unify this and make it more generic.
```
Comment (zabbal) [inline src/libaktualizr/package_manager/packagemanagerfake.cc:113]: Perhaps explicitly saying unification with what exactly is desired would be helpful in future? Like function name or file for example.

File: src/libaktualizr/uptane/imagerepository.cc (hunk 1 above)
Comment (patrickvacek) [reply src/libaktualizr/uptane/imagerepository.cc:118]: Yes, I believe so, although the real tragedy is we don't think we make this particularly obvious in the web UI.

File: src/libaktualizr/package_manager/packagemanagerfake.cc (hunk 2 above)
Comment (patrickvacek) [reply src/libaktualizr/package_manager/packagemanagerfake.cc]: Fair point, done.

File: src/libaktualizr/uptane/imagerepository.cc (hunk 1 above)
Comment (zabbal) [reply src/libaktualizr/uptane/imagerepository.cc:118]: Hmm... I though there's a ticket for that but can't find it now.

**Pull request merged**